autoload/gitgutter/hunk.vim

```diff
@@ -310,14 +310,18 @@ function! s:preview(hunk_diff)
 
   let b:hunk_header = header
 
-  setlocal noreadonly modifiable filetype=diff buftype=nofile bufhidden=delete noswapfile
+  setlocal noreadonly modifiable filetype=diff bufhidden=delete noswapfile
   execute "%delete_"
   call setline(1, body)
   normal! gg
 
-  cnoreabbrev <buffer> <expr> w  getcmdtype() == ':' && getcmdline() == 'w'  ? 'GitGutterStageHunk' : 'w'
-  " Staging hunk from the preview window closes the window anyway.
-  cnoreabbrev <buffer> <expr> wq getcmdtype() == ':' && getcmdline() == 'wq' ? 'GitGutterStageHunk' : 'wq'
+  exe 'file gitgutter://hunk-'.bufnr('%')
```

Comment:
Given the filename that it has now, this could be checked in https://github.com/blueyed/vim-gitgutter/blob/944181f1528887051daf37a0781ad7c2aac97634/autoload/gitgutter/hunk.vim#L195 (instead of "if &previewwindow"), to handle the upcoming nvim floating window there also maybe.

Reply:
Used an explicit buffer-local variable for this now.  Easier to than matching the bufnr from the bufname.. ;)